### DISCUSSION TITLE
How to handle mutable objects with flux?

### DISCUSSION THREAD

**Discussion by TrAnn3l** [fluxcd/flux2#1987] [Q&A]
Hey,
I'm currently struggling to handle mutable objects the gitops way.
The problem is as follows:

- i have a configmap i deployed with flux
- i mount that configmap into a pod
- the pod updates the configmap, which is totally valid
- flux overwrites the updated configmap with its inital content after a reconcile

Whats the best approach to tackle this problem?
How do i tell flux, that it only needs to apply the ressource if it doen't already exists?

**Comment by stefanprodan** [comment]
This is not supported. In the future, if your app uses Kubernetes server-side apply and sets the managed fields on the ConfigMap, Flux could detect the conflict and accept the in-cluster changes.
